Compilation errors on Visual Studio

Compilation fails on Windows. We need a CI setup for that. :)
```
"cl" "-Itests\libgul-test.exe.p" "-Itests" "-I..\..\..\tests" "-Iinclude" "-I..\..\..\include" "-Isrc" "-I..\..\..\src" "/MDd" "/nologo" "/showIncludes" "/utf-8" "/W4" "/EHsc" "/std:c++14" "/permissive-" "/Od" "/Zi" "/Fdtests\libgul-test.exe.p\test_variant.cc.pdb" /Fotests/libgul-test.exe.p/test_variant.cc.obj "/c" ../../../tests/test_variant.cc
C:\Data\Develop\src\doocs\library\common\gul14\include\gul14/variant.h(716): error C2653: 'lib': is not a class or namespace name
C:\Data\Develop\src\doocs\library\common\gul14\include\gul14/variant.h(716): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
C:\Data\Develop\src\doocs\library\common\gul14\include\gul14/variant.h(716): error C2988: unrecognizable template declaration/definition
C:\Data\Develop\src\doocs\library\common\gul14\include\gul14/variant.h(716): error C2143: syntax error: missing ';' before '<'
C:\Data\Develop\src\doocs\library\common\gul14\include\gul14/variant.h(716): error C7525: inline variables require at least '/std:c++17'
C:\Data\Develop\src\doocs\library\common\gul14\include\gul14/variant.h(716): error C2238: unexpected token(s) preceding ';'
C:\Data\Develop\src\doocs\library\common\gul14\include\gul14/variant.h(716): error C2059: syntax error: '<'
C:\Data\Develop\src\doocs\library\common\gul14\include\gul14/variant.h(717): error C2334: unexpected token(s) preceding '{'; skipping apparent function body
C:\Data\Develop\src\doocs\library\common\gul14\include\gul14/variant.h(785): error C2760: syntax error: unexpected token '{', expected 'expression'
C:\Data\Develop\src\doocs\library\common\gul14\include\gul14/variant.h(751): error C7510: 'push_back_t': use of dependent template name must be prefixed with 'template'
C:\Data\Develop\src\doocs\library\common\gul14\include\gul14/variant.h(748): note: This diagnostic occurred in the compiler generated function 'auto gul14::detail_variant::visitation::base::make_fmatrix_impl<F,Vs...>::impl(Is,std::integer_sequence<size_t,_Jx...>,Ls...)'
C:\Data\Develop\src\doocs\library\common\gul14\include\gul14/variant.h(759): error C2760: syntax error: unexpected token '{', expected 'expression'
C:\Data\Develop\src\doocs\library\common\gul14\include\gul14/variant.h(756): note: This diagnostic occurred in the compiler generated function 'auto gul14::detail_variant::visitation::base::make_fmatrix(void)'
C:\Data\Develop\src\doocs\library\common\gul14\include\gul14/variant.h(790): error C2760: syntax error: unexpected token '{', expected 'expression'
C:\Data\Develop\src\doocs\library\common\gul14\include\gul14/variant.h(783): note: This diagnostic occurred in the compiler generated function 'unknown-type gul14::detail_variant::visitation::base::make_fdiagonal(void)'
[37/66] Compiling C++ object tests/libgul-test.exe.p/test_to_number.cc.obj
C:\Data\Develop\src\doocs\library\common\gul14\include\gul14\optional.h(463) : warning C4702: unreachable code
C:\Data\Develop\src\doocs\library\common\gul14\include\gul14\optional.h(463) : warning C4702: unreachable code
C:\Data\Develop\src\doocs\library\common\gul14\include\gul14\optional.h(463) : warning C4702: unreachable code
[40/66] Compiling C++ object tests/lib_ignore_me.a.p/meson-generated_.._incl_date.cc.obj
ninja: build stopped: subcommand failed.
Finished. Build directory: build\x64-windows\debug
```

--- a/include/gul14/variant.h
+++ b/include/gul14/variant.h
@@ -114,6 +114,17 @@ struct array {
 template <typename T, bool>
 struct dependent_type : T {};
 
+template <typename Is, std::size_t J>
+struct push_back;
+
+template <typename Is, std::size_t J>
+using push_back_t = typename push_back<Is, J>::type;
+
+template <std::size_t... Is, std::size_t J>
+struct push_back<std::index_sequence<Is...>, J> {
+    using type = std::index_sequence<Is..., J>;
+};
+
 #if __has_builtin(__type_pack_element) && !(defined(__ICC))
     template <std::size_t I, typename... Ts>
     using type_pack_element_t = __type_pack_element<I, Ts...>;
@@ -730,7 +741,7 @@ using is_nothrow_swappable =
         }
 
         template <typename T, std::size_t N, typename... Is>
-        inline static constexpr const lib::remove_all_extents_t<T> &at(
+        inline static constexpr const std::remove_all_extents_t<T> &at(
             const detail_variant::array<T, N> &elems, std::size_t i, Is... is) noexcept {
           return at(elems[i], is...);
         }
@@ -765,7 +776,7 @@ using is_nothrow_swappable =
           inline static constexpr auto impl(Is,
                                             std::index_sequence<Js...>,
                                             Ls... ls) {
-            return make_farray(impl(lib::push_back_t<Is, Js>{}, ls...)...);
+            return make_farray(impl(detail_variant::push_back_t<Is, Js>{}, ls...)...);
           }
         };
 
@@ -773,7 +784,7 @@ using is_nothrow_swappable =
         inline static constexpr auto make_fmatrix() {
           return make_fmatrix_impl<F, Vs...>::impl(
               std::index_sequence<>{},
-              lib::make_index_sequence<std::decay_t<Vs>::size()>{}...);
+              std::make_index_sequence<std::decay_t<Vs>::size()>{}...);
         }
 
         template <typename F, typename... Vs>
@@ -799,12 +810,12 @@ using is_nothrow_swappable =
         template <typename F, typename V, typename... Vs>
         inline static constexpr auto make_fdiagonal()
             -> decltype(make_fdiagonal_impl<F, V, Vs...>::impl(
-                lib::make_index_sequence<std::decay_t<V>::size()>{})) {
+                std::make_index_sequence<std::decay_t<V>::size()>{})) {
           static_assert(detail_variant::all<(std::decay_t<V>::size() ==
                                   std::decay_t<Vs>::size())...>::value,
                         "all of the variants must be the same size.");
           return make_fdiagonal_impl<F, V, Vs...>::impl(
-              lib::make_index_sequence<std::decay_t<V>::size()>{});
+              std::make_index_sequence<std::decay_t<V>::size()>{});
         }
 #endif
       };

